allensdk/brain_observatory/behavior/behavior_ophys_api/behavior_ophys_nwb_api.py

```diff
@@ -181,9 +194,30 @@ def get_metadata(self) -> dict:
         data['reporter_line'] = list(nwb_subject.reporter_line)
         data['driver_line'] = list(nwb_subject.driver_line)
 
-        experiment_datetime = metadata_nwb_obj.experiment_datetime
-        data['experiment_datetime'] = OphysBehaviorMetadataSchema().load({'experiment_datetime': experiment_datetime}, partial=True)['experiment_datetime']            
-        data['behavior_session_uuid'] = uuid.UUID(data['behavior_session_uuid'])
+        # Add pyNWB OpticalChannel and ImagingPlane metadata to behavior ophys
+        # session metadata
+        try:
+            ophys_module = self.nwbfile.processing['ophys']
+            image_seg = ophys_module.data_interfaces['image_segmentation']
+            imaging_plane = image_seg.plane_segmentations['cell_specimen_table'].imaging_plane
+            optical_channel = imaging_plane.optical_channel[0]
+
+            data['ophys_frame_rate'] = imaging_plane.imaging_rate
+            data['indicator'] = imaging_plane.indicator
+            data['targeted_structure'] = imaging_plane.location
+            data['excitation_lambda'] = imaging_plane.excitation_lambda
+            data['emission_lambda'] = optical_channel.emission_lambda
+        except KeyError:
```

Comment:
Hmm... Nit but maybe this should be raised a bit earlier. Since I could imagine there could be other keyerrors (e.g. perhaps line 201

Reply:
Good catch! I've updated that block so that it uses a `try - except - else` so that the caught exception is specific.